lib/cocoapods/executable.rb

```diff
@@ -93,6 +94,28 @@ def self.popen3(bin, command, stdout, stderr)
       end
     end
 
+    def self.read(input, output)
+      Thread.new do
+        buf = ''
+        begin
+          loop do
+            s = input.readpartial(4096)
```

Comment:
Sure – and in the old code you put it back together. Not necessary with `rpartition` – you output append all newlines and text including the last newline.